tests/resources/test_collection.py

```diff
@@ -303,3 +305,153 @@ async def test_get_collections_search(
         "/collections",
     )
     assert len(resp.json()["collections"]) == 2
+
+
+@requires_pgstac_0_9_2
```

Comment:
This will skip the test for old version of pgstac

IMO it will be a bit **bold** to restrict pgstac>=0.9.2 just because of the collection-pagination issue. 

We will need to add a note in the documentation stating that collection-search is `partially` supported for `0.8.*`